kak-lsp.toml

```diff
@@ -69,6 +69,21 @@ roots = ["package.json", ".git", ".hg"]
 command = "vscode-css-languageserver"
 args = ["--stdio"]
 
+
+# See https://scalameta.org/metals/docs/integrations/new-editor
+[language_server.metals]
+filetypes = ["scala", "sbt", "sc"]
```

Comment:
I'm not sure what `sbt` and `sc` are.
In general we only take filetypes that are also in Kakoune.
So if the plan is to add them to Kakoune's `scala.kak` then this is fine.
(Or the languages are the same maybe they should all get the `scala` filetype).

By default we send the filetype as LSP languageId.
So if metals works better/worse with "sbt" instead of "scala" that would be interesting to know.

Reply:
Aha, got it, didn't know `filetypes` are in the Kakoune terminology.

Assumed this field is for general `*.<file extension>` file extensions.

`scala`, `sbt` and `sc` are all Scala files just with different extension names (`sbt` is for [sbt](https://www.scala-sbt.org/index.html) and `sc` for [ScalaCLI](https://scala-cli.virtuslab.org/)).

Done!

Reply:
ok then we might want

```diff
diff --git a/rc/filetype/scala.kak b/rc/filetype/scala.kak
index 0822a016b..05d7b28ee 100644
--- a/rc/filetype/scala.kak
+++ b/rc/filetype/scala.kak
@@ -4,7 +4,7 @@
 # Detection
 # ‾‾‾‾‾‾‾‾‾
 
-hook global BufCreate .*[.](scala) %{
+hook global BufCreate .*[.](scala|sbt|sc) %{
     set-option buffer filetype scala
 }
 ```